uaclient/entitlements/base.py

```diff
@@ -416,6 +416,9 @@ def can_enable(self) -> Tuple[bool, Optional[CanEnableFailure]]:
     def enable(
         self,
         silent: bool = False,
+        enable_required_services: Optional[bool] = None,
+        disable_incompatible_services: Optional[bool] = None,
+        api: bool = False,
```

Comment:
I know this all be refactored in the future, but what if right now, instead of introducing new parameters to the method, we create two additional methods (names are just a suggestion):

* `handle_can_enable`
* `handle_required_components`

Then, we would just need to call those method directly on the `enable` API function without changing too much the main `enable` method. I believe this is better because this solution will not cross the boundaries between the API and the current enable logic.

But I again understand that this all be refactored in the future (even if we adopt my idea), so feel to disagree with this approach.